vtkext/private/module/vtkF3DRenderer.cxx

```diff
@@ -1560,6 +1651,108 @@ void vtkF3DRenderer::ResetCameraClippingRange()
   this->GridActor->SetUseBounds(gridUseBounds);
 }
 
+vtkBoundingBox vtkF3DRenderer::ComputeVisiblePropOrientedBounds(const vtkMatrix4x4* matrix)
+{
+  const auto isMatrixAxisAligned = [](const vtkMatrix4x4* m, const double tol = 1e-8)
+  {
+    for (size_t i = 0; i < 3; ++i)
+    {
+      size_t nonzerosI = 0;
+      size_t nonzerosJ = 0;
+      for (size_t j = 0; j < 3; ++j)
+      {
+        if (::abs(m->Element[i][j]) > tol)
+          nonzerosI++;
+        if (::abs(m->Element[j][i]) > tol)
+          nonzerosJ++;
+      }
+      if (nonzerosI > 1 || nonzerosJ > 1)
+        return false;
+    }
+    return true;
+  };
+
+  /* Use `PokeMatrix` around the call to `GetBounds()` to extend box.
+   * Only gives the thightest bounds if the transformation is axis-aligned. */
+  const auto extendBoxAxisAligned = [&](vtkProp3D* prop3d, vtkBoundingBox& box)
+  {
+    vtkNew<vtkMatrix4x4> tmpMatrix;
+    vtkMatrix4x4::Multiply4x4(matrix, prop3d->GetMatrix(), tmpMatrix);
+    prop3d->PokeMatrix(tmpMatrix);
+
+    box.AddBounds(prop3d->GetBounds());
+
+    prop3d->PokeMatrix(nullptr);
+  };
+
+  /* Use custom logic to extend box.
+   * Should give the tightest bounds even when non-axis-aligned */
+  const auto extendBoxArbitrary = [&](vtkProp3D* prop3d, vtkBoundingBox& box)
+  {
+    vtkActor* actor = dynamic_cast<vtkActor*>(prop3d);
+    if (actor)
+    {
+      vtkPolyDataMapper* polyMapper = dynamic_cast<vtkPolyDataMapper*>(actor->GetMapper());
+      if (polyMapper)
+      {
+        vtkNew<vtkTransform> t;
+        t->Concatenate(*matrix->Element);
+        t->Concatenate(actor->GetMatrix());
```

Comment:
Again, let's avoid `vtkTransform`. Just multiply the matrices to combine them.  
If the matrices can store a translation, you cannot use a 3x3 matrix and stick with the 4x4 matrix.  
In this case, you also need to use vec4, so make sure to use w=1 for points and w=0 for the fourth component.